packages/react-components/react-field/src/components/Field/Field.types.ts

```diff
@@ -1,17 +1,101 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { Label } from '@fluentui/react-label';
+import type { ComponentProps, ComponentState, Slot, SlotClassNames } from '@fluentui/react-utilities';
+import type { SlotComponent } from './SlotComponent.types';
 
-export type FieldSlots = {
-  root: Slot<'div'>;
+/**
+ * The minimum requirement for a component used by Field.
+ *
+ * Note: the use of VoidFunctionComponent means that component is not *required* to have a children prop,
+ * but it is still allowed to have a children prop.
+ */
+export type FieldComponent = React.VoidFunctionComponent<
+  Pick<
+    React.HTMLAttributes<HTMLElement>,
+    'id' | 'className' | 'style' | 'aria-labelledby' | 'aria-describedby' | 'aria-invalid' | 'aria-errormessage'
+  >
+>;
+
+/**
+ * Slots added by Field
+ */
+export type FieldSlots<T extends FieldComponent> = {
+  root: NonNullable<Slot<'div'>>;
+
+  /**
+   * The underlying component wrapped by this field.
+   *
+   * This is the PRIMARY slot: all intrinsic HTML properties will be applied to this slot,
+   * except `className` and `style`, which remain on the root slot.
+   */
+  field: SlotComponent<T>;
+
+  /**
+   * The label associated with the field.
+   */
+  label?: Slot<typeof Label>;
+
+  /**
+   * A message about the validation state. The appearance of the `validationMessage` depends on `validationState`.
+   */
+  validationMessage?: Slot<'span'>;
+
+  /**
+   * The icon associated with the `validationMessage`. If the `validationState` prop is set, this will default to an
+   * icon corresponding to that state.
+   *
+   * This will only be displayed if `validationMessage` is set.
+   */
+  validationMessageIcon?: Slot<'span'>;
+
+  /**
+   * Additional hint text below the field.
+   */
+  hint?: Slot<'span'>;
 };
 
 /**
  * Field Props
  */
-export type FieldProps = ComponentProps<FieldSlots> & {};
+export type FieldProps<T extends FieldComponent> = ComponentProps<Partial<FieldSlots<T>>, 'field'> & {
+  /**
+   * The orientation of the label relative to the field component.
+   * This only affects the label, and not the validationMessage or hint (which always appear below the field component).
+   *
+   * @default vertical
+   */
+  orientation?: 'vertical' | 'horizontal';
```

Comment:
This comment on the spec review covers some of the reasoning behind the current name: https://github.com/microsoft/fluentui/pull/24372#discussion_r949665227

The main reason not to include "label" in the name is that it affects layout even if there is no label. But that's a bit of an edge case since most/all fields should have labels. I also prefer shorter, one-word prop names when possible.